dotCMS/src/main/java/com/dotcms/content/elasticsearch/util/DotRestHighLevelClientProvider.java

```diff
@@ -0,0 +1,354 @@
+        final String esAuthType = getESAuthType();
+
+        final RestClientBuilder clientBuilder = RestClient
+                .builder(new HttpHost(Config.getStringProperty("ES_HOSTNAME", "127.0.0.1"),
+                        Config.getIntProperty("ES_PORT", 9200), "https"))
+                .setHttpClientConfigCallback((httpClientBuilder) -> {
+                    if (sslContextFromPem != null) {
+                        httpClientBuilder
+                                .setSSLContext(sslContextFromPem);
+                    }
+                    httpClientBuilder
+                            .setDefaultCredentialsProvider(credentialsProvider);
+                    return httpClientBuilder;
+                })
+                .setFailureListener(new RestClient.FailureListener() {
+                    public void onFailure(Node node) {
+                        Logger.error(this, node.toString());
+                    }
+                });
+
+        //Loading token if JWT authentication is set
+        if (esAuthType.equals(JWT_AUTH_TYPE)) {
+            clientBuilder.setDefaultHeaders(new Header[]{new BasicHeader("Authorization",
+                    "Bearer " + Config.getStringProperty("ES_AUTH_JWT_TOKEN", null))});
+            Logger.info(DotRestHighLevelClientProvider.class,
+                    "Initializing Elastic RestHighLevelClient using JWT authentication");
+        }
+
+        return clientBuilder;
+    }
+
+    private String getESAuthType() {
+        return Config.getStringProperty("ES_AUTH_TYPE", BASIC_AUTH_TYPE);
+    }
+
+    private void loadTLSCertificates() throws IOException, GeneralSecurityException {
+        String clientCertPath = getCertPath("ES_AUTH_TLS_CLIENT_CERT", "elasticsearch.pem");
+        String clientKeyPath = getCertPath("ES_AUTH_TLS_CLIENT_KEY", "elasticsearch.key");
+        String serverCertPath = getCertPath("ES_AUTH_TLS_CA_CERT", "root-ca.pem");
+
+        sslContextFromPem = SSLContexts
+                .custom()
+                .loadKeyMaterial(PemReader
+                        .loadKeyStore(
+                                Paths.get(clientCertPath).toFile(),
+                                Paths.get(clientKeyPath).toFile(),
+                                Optional.empty()), "".toCharArray())
+                .loadTrustMaterial(PemReader.loadTrustStore(
+                        Paths.get(serverCertPath).toFile()), null)
+                .build();
+        Logger.info(DotRestHighLevelClientProvider.class,
+                "Initializing Elastic RestHighLevelClient using TLS certificates");
+    }
+
+    @NotNull
+    private String getCertPath(final String propertyName, final String fileName) throws IOException {
+        String clientCertPath = Config.getStringProperty(propertyName, null);
+        if (clientCertPath == null || !Files.exists(Paths.get(clientCertPath))) {
+
+            String assetsRealPath = Config.getStringProperty("ASSET_REAL_PATH",
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'assetsRealPath' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=3712629)